### ISSUE TITLE
Unable to perform the initial migrate of a djangocms project when djangocms_blog is included.

### ISSUE THREAD

**Issue by bxm156** [nephila/djangocms-blog#207]
Django cms blog == 0.6.3
djangocms==3.2.1
django==1.9.2

On a fresh project, you cannot run the initial python manage.my migrate for DjangoCMS when djangocms_blog is included in the INSTALLED_APPS. It looks similar to an issue with the django debug toolbar:
https://github.com/django-debug-toolbar/django-debug-toolbar/issues/457
https://github.com/divio/django-cms/issues/2561

In those cases, the issue had to be fixed in the debug toolbar code, so I'm guessing a similar fix may need to be done in the cms blog code.

On my local dev, I wanted to start from a fresh database. So i deleted my existing one and ran the migrate command, I expected it to create all the tables, instead it failed with the following message:
```
python manage.py makemigrations                                          
/dir/site-packages/cms/forms/fields.py:28: RemovedInDjango110Warning: `PageSelectFormField._has_changed` method should be renamed `has_changed`.
  class PageSelectFormField(forms.MultiValueField):

Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/dir/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/dir/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/dir/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/dir/site-packages/django/core/management/base.py", line 398, in execute
    self.check()
  File "/dir/site-packages/django/core/management/base.py", line 426, in check
    include_deployment_checks=include_deployment_checks,
  File "/dir/site-packages/django/core/checks/registry.py", line 75, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/dir/site-packages/django/core/checks/urls.py", line 10, in check_url_config
    return check_resolver(resolver)
  File "/dir/site-packages/django/core/checks/urls.py", line 19, in check_resolver
    for pattern in resolver.url_patterns:
  File "/dir/site-packages/django/utils/functional.py", line 33, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/dir/site-packages/django/core/urlresolvers.py", line 417, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/dir/site-packages/django/utils/functional.py", line 33, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/dir/site-packages/django/core/urlresolvers.py", line 410, in urlconf_module
    return import_module(self.urlconf_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/Users/bryanmarty/queercon/queercon/urls.py", line 27, in <module>
    url(r'^admin/', include(admin.site.urls)),
  File "/dir/site-packages/django/contrib/admin/sites.py", line 303, in urls
    return self.get_urls(), 'admin', self.name
  File "/dir/site-packages/django/contrib/admin/sites.py", line 287, in get_urls
    url(r'^%s/%s/' % (model._meta.app_label, model._meta.model_name), include(model_admin.urls)),
  File "/dir/site-packages/django/contrib/admin/options.py", line 561, in urls
    return self.get_urls()
  File "/dir/site-packages/cms/admin/pageadmin.py", line 151, in get_urls
    url_patterns += super(PageAdmin, self).get_urls()
  File "/dir/site-packages/cms/admin/placeholderadmin.py", line 164, in get_urls
    from cms.urls import SLUG_REGEXP
  File "/dir/site-packages/cms/urls.py", line 20, in <module>
    urlpatterns = get_app_patterns()
  File "/dir/site-packages/cms/appresolver.py", line 209, in get_app_patterns
    current_site = Site.objects.get_current()
  File "/dir/site-packages/django/contrib/sites/models.py", line 65, in get_current
    return self._get_site_by_id(site_id)
  File "/dir/site-packages/django/contrib/sites/models.py", line 35, in _get_site_by_id
    site = self.get(pk=site_id)
  File "/dir/site-packages/django/db/models/manager.py", line 122, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/dir/site-packages/django/db/models/query.py", line 381, in get
    num = len(clone)
  File "/dir/site-packages/django/db/models/query.py", line 240, in __len__
    self._fetch_all()
  File "/dir/site-packages/django/db/models/query.py", line 1074, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/dir/site-packages/django/db/models/query.py", line 52, in __iter__
    results = compiler.execute_sql()
  File "/dir/site-packages/django/db/models/sql/compiler.py", line 848, in execute_sql
    cursor.execute(sql, params)
  File "/dir/site-packages/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/dir/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/dir/site-packages/django/db/utils.py", line 95, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/dir/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/dir/site-packages/django/db/backends/sqlite3/base.py", line 323, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.OperationalError: no such table: django_site
```

I was able to do the initial  migration by removing djangocms_blog from the INSTALLED_APPS, and then adding it back and running migrate a second time.

**Comment by yakky** [comment]
@bxm156 this is actually a bug in django-cms . Is fixed in divio/django-cms#5012 which will be released in upcoming 3.2.2
This bug is triggered when any apphook-enabled application is added in INSTALLED_APPS

**Comment by bxm156** [comment]
Oh, good to know. Thanks!
